### ISSUE TITLE
SharedProps returning error when used in usePage hook

### ISSUE THREAD

**Issue by josevelaz** [adonisjs/inertia#46]
### Package version

1.2.2

### Describe the bug

When augmenting the SharedProps interface like shown in [the documentation](https://docs.adonisjs.com/guides/views-and-templates/inertia#shared-props), an error is raised by the `usePage<SharedProps>()` hook.

```
 tsserver: Type 'SharedProps' does not satisfy the constraint 'PageProps'.
   Index signature for type 'string' is missing in type 'SharedProps'. [2344]
```

Passing any value into the module augmentation causes it to break

```
declare module '@adonisjs/inertia/types' {
  export interface SharedProps extends InferSharedProps<typeof inertiaConfig> {
    props2: string
  }
}
```

If prop2: string is removed then the error above goes away

<img width="1542" alt="image" src="https://github.com/user-attachments/assets/a5ef8a95-4046-45ff-9cb9-0b6964c4ef3b">

![CleanShot 2024-11-19 at 10 52 14@2x](https://github.com/user-attachments/assets/903f509b-9d5d-48f1-97dc-6eb6179538c4)

**Comment by josevelaz** [comment]
@Julien-R44 reaching out making sure you were able to replicate it?
I was able to replicate it off a new adonisjs project so it seems like its some broken functionality. Happening on v2 aswell, although im aware not much changed from v1 -> v2

**Comment by Julien-R44** [comment]
Yup, can confirm this is a bug. I haven't had time to look into it yet. Happy to accept a PR

**Comment by Barabasbalazs** [comment]
I think that by rewriting the template like this in the `config/inerta.ts` file, 

```TypeScript
declare module '@adonisjs/inertia/types' {
  export type SharedProps = InferSharedProps<typeof inertiaConfig> & {
    prop2: string
  }
}
```

and also removing this interface declaration from the `types` file can solve this issue:

```TypeScript
export interface SharedProps {}
```

See:

![inertia_shared_prop_types](https://github.com/user-attachments/assets/4ab4a2e7-a025-464c-81b5-455a18d8aa2e)

**Comment by Julien-R44** [comment]
Fixed by #50

**Comment by Julien-R44** [comment]
Finally #50 wasn't enough, see https://github.com/adonisjs/inertia/pull/53

I reverted it until we found another solution. So, re-opening this issue

**Comment by Barabasbalazs** [comment]
@Julien-R44 I checked the implementation that I did and it seems to work for `InferPageProps` as well. Can you maybe provide some detail about how this PR broke that?

![inferpageprops2](https://github.com/user-attachments/assets/6c3b632b-f99a-4883-83ab-e18db5b7b5f0)

**Comment by Julien-R44** [comment]
You need to configure InferPageProps correctly : point to a real controller that returns real data. For example, you could create this controller: 

```ts
export class AboutController {
  async index({ inertia }: HttpContext) {
    return inertia.render('about', { bar: 42 })
  }
}
```

If you do this then you will be unable to access `props.bar`. 
See https://docs.adonisjs.com/guides/views-and-templates/inertia#types-sharing

**Comment by Barabasbalazs** [comment]
@Julien-R44 That's what I did. 

![about_controller](https://github.com/user-attachments/assets/d476ea2f-3c6c-4022-b942-32f881d0f860)

The type checking seems to be working but it looks like the props don't get passed properly to it from the controller. I don't think that this was the PR that broke it.
You can check out this [repo](https://github.com/Barabasbalazs/react_inertia) in order to reproduce the issue.

**Comment by corentinclichy** [comment]
I have the same issue. Is there a way to fix that ?

**Comment by thetutlage** [comment]
Should be fixed in the next major release. https://github.com/adonisjs/inertia/releases/tag/v4.0.0-next.1
